lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -334,9 +337,17 @@ struct ExpressionVisitor
     newInfo.stream.indent(newInfo.indentLevel);
     if (failed(printExpression(worklist, newInfo)))
       return failure();
+
     info.stream << info.valueMap.lookup(yieldedValue);
-    for (unsigned k = 0; k < newInfo.openParens + 1; ++k)
+
+    for(int j=0;j<newInfo.openParens;++j)
+      info.stream <<")";
+
+    if(weight != 0){
+      info.stream <<" :weight "<< weight << "))";
+    } else{
       info.stream << ")";
+    }
```

Comment:
Nit: If statements where the body of all branches is just one line should omit the curly braces according to the LLVM coding style guide

Reply:
We could also remove one of the two closing parenthesis at the end of the statement in the then-branch and move the statement in the else-branch out such that the if-statement only has a then-statement.